environment.yml

```diff
@@ -23,7 +23,7 @@ dependencies:
   - pandas-stubs
 
   # pyspark extra
-  - pyspark[connect] >= 3.2.0
+  - pyspark[connect] >= 3.2.0, < 4.0.0
```

Comment:
Sadly, neither `~= 3.2` nor `>= 3.2, == 3.*` plays well with the `generate_pip_deps_from_conda` script.